src/platform/libretro/libretro.c

```diff
@@ -366,6 +372,86 @@ bool retro_load_game(const struct retro_game_info* game) {
 	_reloadSettings();
 	GBAContextLoadROMFromVFile(&context, rom, save);
 	GBAContextStart(&context);
+	
+	memset(descs, 0, sizeof(descs));
+	savedata_size = retro_get_memory_size(RETRO_MEMORY_SAVE_RAM);
+	
+	/* Map internal working RAM */
+	descs[0].ptr    =  gba->memory.iwram;
+	descs[0].start  =  BASE_WORKING_IRAM;
+	descs[0].len    =  SIZE_WORKING_IRAM;
+	descs[0].select = 0xFF000000; /* From vbam-libretro */
+	
+	/* Map working RAM */
+	descs[1].ptr    = gba->memory.wram;
+	descs[1].start  = BASE_WORKING_RAM;
+	descs[1].len    = SIZE_WORKING_RAM;
+	descs[1].select = 0xFF000000;
+	
+	/* Map save RAM */
+	descs[2].ptr    = savedata_size ? savedata : NULL;
+	descs[2].start  = BASE_CART_SRAM;
+	descs[2].len    = savedata_size;
+	descs[2].select = 0xFF000000;
+	
+	/* Map ROM */
+	descs[3].ptr    = gba->memory.rom;
+	descs[3].start  = BASE_CART0;
+	descs[3].len    = SIZE_CART0;
+	descs[3].select = 0xFF000000;
+	descs[3].flags  = RETRO_MEMDESC_CONST;
+	
+	descs[4].ptr    = gba->memory.rom;
+	descs[4].start  = BASE_CART1;
+	descs[4].len    = SIZE_CART1;
+	descs[4].select = 0xFF000000;
+	descs[4].offset = SIZE_CART0;
```

Comment:
Exposing the mirrors is the correct thing to do, but yeah, if it's a mirror, then offset should be zero.